### PR TITLE
Reset existing players when they join new room

### PR DESCRIPTION
**Pull request by DanielMuckelbauer**
Right now, an inactive player or a player with a choice will have those same properties for the new room. Also, they will still be related to the old room they were in.

### CODE REVIEW

File: src/app/room/components/room.component.ts [hunk 1]
```diff
@@ -44,15 +44,28 @@ export class RoomComponent implements OnInit {
...
+        this.resetExistingPlayer(roomId);
       }
     });
   }
 
+  private createNewPlayer(roomId: string): void {
+    this.playerCreationService.createPlayer(roomId).subscribe(_ => this.loadingStatus.isLoading = false);
+  }
+
+  private resetExistingPlayer(roomId: string): void {
```
Comment (imeha) [inline src/app/room/components/room.component.ts:61]: Maybe also if the roomId(from the request parameter) !== roomId(in the player document).

File: src/app/room/components/room.component.ts [hunk 2]
```diff
@@ -44,15 +44,28 @@ export class RoomComponent implements OnInit {
...
+    this.firebasePlayerService.updateCurrent({
+      room: roomId,
+      isObserver: false,
+      choice: null,
+    } as Partial<Player>).subscribe(() => this.loadingStatus = { isLoading: false });
```
Comment (imeha) [inline src/app/room/components/room.component.ts:66]: Can you add `this.loadingStatus = { isLoading: false }` to `finalize` in the parent method, please?

File: src/app/room/components/room.component.ts (hunk 1 above)
Comment (DanielMuckelbauer) [reply src/app/room/components/room.component.ts]: Good point, done

File: src/app/room/components/room.component.ts (hunk 2 above)
Comment (DanielMuckelbauer) [reply src/app/room/components/room.component.ts]: Refactored it all a little, still not perfect but much better I think, check it out

**Review by imeha — APPROVED** [review]

**Pull request merged**